src/server/saved_objects/service/lib/decorate_es_error.js

```diff
@@ -51,36 +45,47 @@ export function decorateEsError(error) {
 
   const { reason } = get(error, 'body.error', {});
   if (
-    error instanceof ConnectionFault ||
-    error instanceof ServiceUnavailable ||
-    error instanceof NoConnections ||
-    error instanceof RequestTimeout
+    error instanceof ConnectionError ||
+    error instanceof NoLivingConnectionsError ||
+    error instanceof TimeoutError ||
+    // TODO: The new client considers as "service unavailable"
+    // (so worth of request retry) errors with status codes 502, 503, and 504.
+    // Do we want to handle them as well?
+    error.statusCode === 503 // Service Unavailable
```

Comment:
_TODO:_ The new client considers as "service unavailable" (so worth of request retry) errors with status codes 502, 503, and 504, we need to handle them as well.